### ISSUE TITLE
Best way to handle a self updating node module?

### ISSUE THREAD

**Issue by ajbdev** [electron-userland/electron-builder#2202]
I have a node module that self updates its javascript source occasionally based on some external factors. Currently, when this happens I have to repackage a completely new version of the app and redistribute the entire thing in order for the change to take effect. I'd like to automate this so the module can update itself on the client side so I do not need to repackage every single time.

I attempted to use asarUnpack to leave this node module unpacked in the resource directory along with using the library hazardous (https://github.com/epsitec-sa/hazardous) which monkey patches path resolution to seek unpacked files before using the one stored in the ASAR. This works in the main thread, but fails in the renderer thread.

Is there a straight forward way of accomplishing this and having it work in both the renderer and main thread?

**Comment by develar** [comment]
You cannot modify app installation dir. So, keep such node module in the app data dir.

**Comment by ajbdev** [comment]
Thanks, I solved this by using the app data dir. 
It was a little tricky to actually use the javascript in the app data folder. You cannot simply require them and expect it to work - javascript files in the app data folder are basically sandboxed off from all your app code, which was a problem for me because I needed those files to access modules in my ASAR.

Perhaps this is not the most elegeant solution, but what did work was reading the file in as a string and passing it to a function constructor like so:

```
const externalJsCodeAsFunction = new Function('arg1',
      fs.readFileSync(`${app.getPath('appData')}/myExternalJs.js`, err => {
        console.log(err)
      })
)(thingsIneedToPass)
```

This does affect how your external code works - in order to get it to work right I had to convert the external javascript from a normal node module into something that returned a value at the very end so that it worked as function code normally would be. 

If anyone has a better approach I'd love to hear it.
